docs/fundamentals/build-container-images.md

```diff
@@ -0,0 +1,183 @@
+---
+title: Build container images
+description: Learn how to build container images from your .NET Aspire resources.
+ms.date: 07/24/2025
+ai-usage: ai-assisted
+---
+
+# Build container images
+
+.NET Aspire provides powerful APIs for building container images from your resources during publishing and deployment operations. This article covers the key components that enable programmatic container image creation and progress reporting.
+
+## Overview
+
+When you publish or deploy Aspire applications, the platform automatically builds container images for your services. This process involves two main components:
```

Comment:
I think this terminology can be a little misleading. Currently, we only automatically build images when you publish via Docker compose. I might word this generally as, during deploy and publish the container image builder is available to use for compiling images for applicable resources.